neuralprophet/time_net.py

```diff
@@ -188,29 +185,44 @@ def __init__(
         self.compute_components_flag = compute_components_flag
 
         # Optimizer and LR Scheduler
-        self.optimizer = self.config_train.optimizer
-        self.scheduler = self.config_train.scheduler
+        self._optimizer = self.config_train.optimizer
+        self._scheduler = self.config_train.scheduler
+        self.automatic_optimization = False
 
         # Hyperparameters (can be tuned using trainer.tune())
         self.learning_rate = self.config_train.learning_rate if self.config_train.learning_rate is not None else 1e-3
         self.batch_size = self.config_train.batch_size
 
         # Metrics Config
-        self.log_args = {
-            "on_step": False,
-            "on_epoch": True,
-            "prog_bar": True,
-            "batch_size": self.config_train.batch_size,
-        }
-        self.metrics_train = torchmetrics.MetricCollection(metrics=metrics)
-        self.metrics_val = torchmetrics.MetricCollection(metrics=metrics, postfix="_val")
+        if not minimal:
+            self.log_args = {
+                "on_step": False,
+                "on_epoch": True,
+                "prog_bar": True,
+                "batch_size": self.config_train.batch_size,
+            }
+            self.metrics_train = torchmetrics.MetricCollection(metrics=metrics)
+            self.metrics_val = torchmetrics.MetricCollection(metrics=metrics, postfix="_val")
 
         # For Multiple Time Series Analysis
         self.id_list = id_list
         self.id_dict = dict((key, i) for i, key in enumerate(id_list))
         self.num_trends_modelled = num_trends_modelled
         self.num_seasonalities_modelled = num_seasonalities_modelled
 
+        # Regularization
+        self.reg_enabled = utils.check_for_regularization(
+            [
+                config_season,
+                config_regressors,
+                config_lagged_regressors,
+                config_ar,
+                config_events,
+                config_trend,
+                config_holidays,
+            ]
+        )
```

Comment:
Like the idea to check if regularization is enabled only once (instead of calling the regularization function with each iteration and figuring out it was not active) - yet we should only have one way to check if regularization is enabled or not - not multiple ones.